src/client.rs

```diff
@@ -0,0 +1,98 @@
+use crate::error::Error;
+
+use kube::api::{Meta, PatchParams, PostParams};
+use kube::client::Client as KubeClient;
+use kube::Api;
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+/// This `Client` can be used to access Kubernetes.
+/// It wraps an underlying [kube::client::Client] and provides some common functionality.
+#[derive(Clone)]
+pub struct Client {
+    client: KubeClient,
+    post_params: PostParams,
+    patch_params: PatchParams,
+}
+
+impl Client {
+    pub fn new(client: KubeClient) -> Self {
+        Client {
+            client,
+            post_params: PostParams::default(),
```

Comment:
Do we expect at some point to need different sets of params with the same client? If yes, maybe add an Option<...> to the interface already so we do not need to change it later on?

Reply:
I don't think it's too important for post & patch, no. For List maybe. I'd keep it like this for now.